plugins/cost-insights/src/utils/duration.ts

```diff
@@ -94,3 +80,12 @@ export function inclusiveEndDateOf(
 export function intervalsOf(duration: Duration, inclusiveEndDate: string) {
   return `R2/${duration}/${exclusiveEndDateOf(duration, inclusiveEndDate)}`;
 }
+
+function quarterEndDate(inclusiveEndDate: string): string {
+  const endDate = moment(inclusiveEndDate).utc();
+  const endOfQuarter = endDate.endOf('quarter').format(DEFAULT_DATE_FORMAT);
+  if (endOfQuarter === inclusiveEndDate) {
+    return endDate.add(1, 'day').format(DEFAULT_DATE_FORMAT);
+  }
+  return endDate.startOf('quarter').format(DEFAULT_DATE_FORMAT);
+}
```

Comment:
This is the fix for the duplicate months bug, I'm guessing? Could we add some tests for this?